InfiniTAM/InfiniTAM_node.cpp

```diff
@@ -169,24 +184,24 @@ int main(int argc, char** argv) try {
   }
 
   printf("initialising ...\n");
-  ImageSourceEngine* image_source = NULL;
-  IMUSourceEngine* imu_source = NULL;
 
-  CreateDefaultImageSource(arg1, arg2, arg3, arg4, node_handle, image_source,
-                           imu_source);
+  SetUpSources(arg1, arg2, arg3, arg4, node_handle, image_source, imu_source);
```

Comment:
why are you not passing the main_engine? So you could avoid having 
```C++
ITMMainEngine* main_engine_ = NULL;
```
in the `ImageSourceEngine`, I guess.